### PR TITLE
feat(Admin): Ajout d'un filtre pour retrouver les dépôts de besoins incrémentaux

### PR DESCRIPTION
**Pull request by madjid-asa**
### Quoi ?

Ajout d'un filtre pour retrouver les dépôts de besoins incrémentaux.

### Pourquoi ?

Faciliter la vie des admins.

### CODE REVIEW

File: lemarche/tenders/admin.py [hunk 1]
```diff
@@ -54,9 +54,28 @@ def queryset(self, request, queryset):
         return queryset
 
 
-class AmountFilter(MultiChoice):
-    FILTER_LABEL = Tender._meta.get_field("amount").verbose_name
+class AmountCustomFilter(admin.SimpleListFilter):
+    title = "Montant du besoin"
+    # FILTER_LABEL = Tender._meta.get_field("amount").verbose_name
     BUTTON_LABEL = "Appliquer"
```
Comment (raphodn) [inline lemarche/tenders/admin.py:60]: pas l'air d'avoir encore besoin du BUTTON ?

File: lemarche/tenders/models.py [hunk 2]
```diff
@@ -78,7 +107,46 @@ def is_live(self):
...
 
     def has_amount(self):
-        return self.filter(Q(amount__isnull=False) | Q(amount_exact__isnull=False))
+        return self.filter(Q(amount__isnull=False) | Q(amount_exact__isnull=False)).annotate(
+            has_amount_exact=Case(
+                When(amount_exact__isnull=False, then=Value(True)), default=Value(False), output_field=BooleanField()
+            )
+        )
+
+    def filter_by_amount(self, amount: int, operation: str = "gte"):
```
Comment (raphodn) [inline lemarche/tenders/models.py:116]:
```suggestion
    def filter_by_amount_exact(self, amount: int, operation: str = "gte"):
```
pour clarifier, car sinon on pense qu'on doit passer un range (comme ailleurs)

File: lemarche/tenders/tests.py [hunk 3]
```diff
@@ -971,3 +971,38 @@ def test_edit_form_no_matching_on_validate_submission(self):
...
+            find_amount_ranges(5000, "invalid_op")
+
+    def test_edge_case(self):
+        """Test an edge case, such as exactly 1M€ for 'gt' operation."""
+        expected_keys = [tender_constants.AMOUNT_RANGE_1000_MORE]
+        self.assertListEqual(find_amount_ranges(1000000, "gt"), expected_keys)
+
+    def test_no_matching_ranges(self):
+        """Test when no ranges match the criteria."""
+        self.assertListEqual(find_amount_ranges(100, "lte"), [tender_constants.AMOUNT_RANGE_0_1])
```
Comment (raphodn) [inline lemarche/tenders/tests.py:1008]: mettre dans une variable intermédiaire expected_keys ?

File: lemarche/tenders/admin.py [hunk 4]
```diff
@@ -54,9 +54,28 @@ def queryset(self, request, queryset):
...
+        )
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        amount_10k = 10 * 10**3  # 10k
+        if value == "<10k":
+            return queryset.filter_by_amount_exact(amount_10k, operation="lt")
+        elif value == "5k-10k":
+            return queryset.filter_by_amount_exact(amount_10k, operation="lte").filter_by_amount_exact(
+                amount_10k, operation="gte"
```
Comment (raphodn) [inline lemarche/tenders/admin.py:75]: typo ici : je corrige dans #1112
